### ISSUE TITLE
"Return value contains a constant value" spammed many times

### ISSUE THREAD

**Issue by jfecher** [noir-lang/noir#5014]
### Aim

```rs
global COUNT = 400;

fn main() -> pub BoundedVec<[Field; 1], COUNT> {
    let mut result = BoundedVec::new();
    let mut x = 0;

    for _ in 0..COUNT {
        if (x < 1) {
            result.push([0]);
        }
    }

    result
}
```

### Expected Behavior

A "return value contains constant value warning"

### Bug

Many repeated warnings are issued, presumably one for each element of the returned array.

### To Reproduce

1.
2.
3.
4.

### Project Impact

None

### Workaround

None

### Installation Method

None

### Would you like to submit a PR for this Issue?

None

**Comment by TomAFrench** [comment]
Agreed we should deduplicate these for each Noir program. This issue also applies to other instances such as `verify_proof`, etc.

**Comment by TomAFrench** [comment]
https://github.com/noir-lang/noir/issues/4365

**Comment by TomAFrench** [comment]
Closed in #2936

**Comment by jfecher** [comment]
> Closed in https://github.com/noir-lang/noir/issues/2936
That issue seems unrelated. Typo?

**Comment by TomAFrench** [comment]
Yep, I've fixed the link above.
